### DISCUSSION TITLE
Consolidate template config options

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#33119] [Suggest an Idea]
### Tell us more.

Existing options like `branchName`, `branchPrefix`, `commitMessagePrefix`, etc could all be consolidated into a `templates` object, e.g.

```json
{
  "templates": {
    "branchName": "foo",
    "commitMessagePrefix": "bar",
    "dependencyDashboardTitle": "baz"
}
```

We need to think how to handle any _nested_ templates though. We could keep the existing nesting approach:

```json
{
  "templates": {
    "branchName": "foo",
    "commitMessagePrefix": "bar",
    "dependencyDashboardTitle": "baz",
    "npm": {
      "commitMessageTopic" "apples"
    },
    "group": {
      "commitMessagePrefix": "oranges"
    }
}
```

This is a bit messy but maybe more readable than before.

**Comment by secustor** [comment]
Why not simply rename options which are supporting templates with a `Template` suffix. That would be IMO simpler and consistent with other usages of templates like in the custom manager.

**Reply by rarkins** [reply]
It's a possibility. But I was hoping to have them expanded quite a lot (e.g. making dependency dashboard content fully configurable) and ideally grouped together documentation-wise (e.g. on a dedicated docs page)

**Reply by secustor** [reply]
In that case it maybe better for UX to completely drop the `Template` prefix and rather use a field in [`lib/config/options`](https://github.com/renovatebot/renovate/blob/main/lib/config/options/index.ts) do document that the option supports templates.

That being said it makes sense to group options together like `dependencyDashboard.*`.

**Reply by jamietanna** [reply]
We do now document the fields that support templating in https://docs.renovatebot.com/templates/ and have a `supportsTemplating=true`
